ext/curl/curl_url.stub.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+/** @generate-class-entries */
+
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+/**
+ * @strict-properties
+ * @not-serializable
+ */
+final class CurlUrl implements Stringable
+{
+    /**
+     * @var int
+     * @cname CURLU_APPENDQUERY
+     */
+    public const APPENDQUERY = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DEFAULT_PORT
+     */
+    public const DEFAULT_PORT = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DEFAULT_SCHEME
+     */
+    public const DEFAULT_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_DISALLOW_USER
+     */
+    public const DISALLOW_USER = UNKNOWN;
+
+    /**
+     * @var int 
+     * @cname CURLU_GUESS_SCHEME
+     */
+    public const GUESS_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_NO_DEFAULT_PORT
+     */
+    public const NO_DEFAULT_PORT = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_NON_SUPPORT_SCHEME
+     */
+    public const NON_SUPPORT_SCHEME = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_PATH_AS_IS
+     */
+    public const PATH_AS_IS = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_URLDECODE
+     */
+    public const URLDECODE = UNKNOWN;
+
+    /**
+     * @var int
+     * @cname CURLU_URLENCODE
+     */
+    public const URLENCODE = UNKNOWN;
+
+#if LIBCURL_VERSION_NUM >= 0x074300 /* Available since 7.67.0 */
+    /**
+     * @var int
+     * @cname CURLU_NO_AUTHORITY
+     */
+    public const NO_AUTHORITY = UNKNOWN;
+#endif
+      
+#if LIBCURL_VERSION_NUM >= 0x074e00 /* Available since 7.78.0 */
+    /**
+     * @var int
+     * @cname CURLU_ALLOW_SPACE
+     */
+    public const ALLOW_SPACE = UNKNOWN;
+#endif
+
+    public function __construct(?string $url = null, int $flags = 0) {}
+
+    public function get(int $flags = 0): string {}
+    public function set(?string $url, int $flags = 0): CurlUrl {}
+
+    public function getHost(): ?string {}
+    public function setHost(?string $host): CurlUrl {}
+
+    public function getScheme(): ?string {}
+    public function setScheme(?string $scheme, int $flags = 0): CurlUrl {}
+
+    public function getPort(int $flags = 0): ?int {}
+    public function setPort(?int $port): CurlUrl {}
+
+    public function getPath(int $flags = 0): string {}
+    public function setPath(?string $scheme, int $flags = 0): CurlUrl {}
+
+    public function getQuery(int $flags = 0): ?string {}
+    public function setQuery(?string $query, int $flags = 0): CurlUrl {}
+
+    public function getFragment(int $flags = 0): ?string {}
+    public function setFragment(?string $fragment, int $flags = 0): CurlUrl {}
+
+    public function getUser(int $flags = 0): ?string {}
+    public function setUser(?string $user, int $flags = 0): CurlUrl {}
+
+    public function getPassword(int $flags = 0): ?string {}
+    public function setPassword(?string $password, int $flags = 0): CurlUrl {}
+
+    public function getOptions(int $flags = 0): ?string {}
```

Comment:
`options` is used for some protocols like imap, and `ZoneId` is used for IPv6 addresses. Those are documented here : https://curl.se/libcurl/c/curl_url_set.html

Reply:
Ah, searching in the context of IMAP and IPv6 gives better clues, thanks. My main concern was that "options" sounds like it might be options *for curl*, rather than options *embedded in the URL*; but if that's the term curl uses, I won't try to second guess it.